src/cephadm/cephadm

```diff
@@ -3970,8 +3978,9 @@ def command_deploy(ctx):
 
     redeploy = False
     unit_name = get_unit_name(ctx.fsid, daemon_type, daemon_id)
+    container_name = 'ceph-%s-%s.%s' % (ctx.fsid, daemon_type, daemon_id)
     (_, state, _) = check_unit(ctx, unit_name)
-    if state == 'running':
+    if state == 'running' or is_container_running(ctx, container_name):
```

Comment:
`check_unit` returns `(enabled, state, installed)`. What about simplifying this to

```python
    (_, _, installed) = check_unit(ctx, unit_name)
    if installed:
```

Reply:
That sounds right. I'm going to test this and if it works I'll swap to this change.

Reply:
@sebastian-philipp This seems like it should work in most situations. However, it is possible to be in a state where this check passes and it considers this a redeploy but the port needed is actually taken. For example, stop the node-exporter service on a host, use the `nc` tool to listen to node-exporter's port `nc -l localhost 9100` and then go and run a `ceph orch redeploy node-exporter` command. The node-exporter service will fail due to the port being taken but the user will never get the error message specifying this is the issue because the port wasn't checked. By checking for the container being up  I think we can be sure node-exporter is holding the port it needs and redeploy can be done. Now, all of this seems like a pretty unlikely scenario so I'm fine still going with this change if you think this is acceptable.

Reply:
> if you think this is acceptable.

by now, you're the expert in this part of the code

Reply:
I personally prefer checking the container. I think it's a more accurate view of whether we're actually redeploying or not.